functionary/ui/forms/forms.py

```diff
@@ -1,28 +1,63 @@
+import json
+
 from django.forms import (
     BooleanField,
     CharField,
     DateField,
     DateTimeField,
-    DecimalField,
     FloatField,
     Form,
     IntegerField,
     JSONField,
-    TimeField,
+    Textarea,
 )
+from django.forms.widgets import DateInput, DateTimeInput
+
+
+class HTMLDateInput(DateInput):
+    input_type = "date"
+
+
+class HTMLDateTimeInput(DateTimeInput):
+    input_type = "datetime-local"
+
 
 _field_mapping = {
-    "integer": IntegerField,
-    "string": CharField,
-    "float": FloatField,
-    "double": DecimalField,
-    "boolean": BooleanField,
-    "date": DateField,  # TODO set input_formats= on constructor if needed
-    "datetime": DateTimeField,  # TODO set input_formats= on constructor if needed
-    "time": TimeField,  # TODO set input_formats= on constructor if needed
-    "object": JSONField,
+    "integer": (IntegerField, None),
+    "string": (CharField, None),
+    "text": (CharField, Textarea),
+    "number": (FloatField, None),
+    "boolean": (BooleanField, None),
+    "date": (DateField, HTMLDateInput),
+    "date-time": (
+        DateTimeField,
+        HTMLDateTimeInput,
+    ),
+    "json": (JSONField, Textarea),
 }
 
+_transform_intitial_mapping = {"json": json.loads}
+
+
+def get_param_type(param_dict):
```

Comment:
Let's add a simple one line docstring here (and places like it going forward).  Also, for the sake of consistency, if this is not intended to be used outside of this file, let's prepend the name with an underscore.

Reply:
Fixed